### DISCUSSION TITLE
[Unstable] Abnormal item stock count in outpost merchants

### DISCUSSION THREAD

**Discussion by NyanSpaghetti** [FakeFishGames/Barotrauma#13674] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

When revisiting an outpost after a couple missions (~5 missions), some items can restock up to abnormally high amounts. 
For me, the stock count was increased to 46 for all affected items, while other items seemed normal. 
It was possible to purchase all 46 items in stock.
Later, when revisiting another outpost, the stock count for some items will be set to the **same** number (46 in my case).
However, this number can be different for others (such as 36, for someone else who encountered a similar/same issue). 
This affects many (seemingly) random items, such as Fentanyl, Naloxone, Deusizine, Gene Splicer, FPGA Circuit, Dementonite, Physicorium, The Sailorʹs Guide, etc. 
However, I was unable to see this issue in Arms Dealers.
![image 1](https://github.com/FakeFishGames/Barotrauma/assets/47715070/2b61120f-d471-4dcb-b548-2d18737ff4f5)

### Reproduction steps

1. Start a multiplayer campaign in beta refactor_our_souls (V1.2.13.0) on Steam.
2. Revisit an outpost after a couple missions. (For me, the issue happened in coalition outposts (Reputation 100). The issue started happening after: entering the Hydrothermal Wastes right after The Ultimate Jest quest (experienced the known bug where the quest is shown as failed), and then returning back to The Great Sea, past the Clownhaven, into an already visited coalition outpost.)
3. **The exact conditions of reproduction is not known**, but I'm creating this discussion so others can add to this discussion after experiencing a similar issue.

Note: I believe it's unrelated, but it may be worth mentioning that my campaign has had zero Thalamus encounters so far, despite reaching the Hydrothermal Wastes. Due to this, I believe there may be something wrong with my particular save file, potentially after experiencing a crash encountered while loading an outpost.

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.2.13.0 (unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by NotWendy** [comment]
I'll add that the amount of items in stock consistently skyrockets to 100 in loops (NG+). Might be related to the restock mechanic in some way, might also be due to how stock amounts are calculated based on current difficulty. Both in loop 0 and in loop 1+ massive amounts of items being available in shops detract from the game's challenge.

**Comment by BlackMoons** [comment]
Seen this occur in the current stable version too.

**Comment by Regalis11** [comment]
Thank you for the report!

It seems that the stock gradually moves towards the maximum, and turns out that for items with no maximum amount defined, that means they'll eventually increase all the way to 100. Created a ticket about this in the issue tracker: https://github.com/FakeFishGames/Barotrauma/issues/13677

**Comment by Regalis11** [comment]
This has been fixed in our private repo now, coming up in the next unstable build!
